### ISSUE TITLE
Crashes on 32-bit ARM (likely a bug in Metrics dependency)

### ISSUE THREAD

**Issue by mattdsteele** [owncast/owncast#2746]
Running the latest `develop` branch on a Raspberry Pi 4 using an older 32-bit OS [as mentioned in the docs](https://owncast.online/docs/codecs/#raspberry-pi), Owncast will crash after a minute or so. 

```
INFO[2023-02-26T21:09:42-06:00] Inbound stream connected.                    
INFO[2023-02-26T21:09:42-06:00] Video transcoder started using OpenMAX (omx) with 1 stream variants. 
panic: unaligned 64-bit atomic operation
        panic: unaligned 64-bit atomic operation

goroutine 29 [running]:
runtime/internal/atomic.panicUnaligned()
        /usr/local/go/src/runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Load64(0x629424c)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.s:280 +0x14
github.com/nakabonne/tstorage.(*memoryPartition).minTimestamp(...)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/memory_partition.go:144
github.com/nakabonne/tstorage.(*storage).flushPartitions(0x615f980)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/storage.go:475 +0x120
github.com/nakabonne/tstorage.(*storage).Close(0x615f980)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/storage.go:420 +0x12c
panic({0x14b4030, 0x3949ce4})
        /usr/local/go/src/runtime/panic.go:884 +0x218
runtime/internal/atomic.panicUnaligned()
        /usr/local/go/src/runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Load64(0x6294254)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.s:280 +0x14
github.com/nakabonne/tstorage.(*memoryPartition).maxTimestamp(...)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/memory_partition.go:148
github.com/nakabonne/tstorage.(*memoryPartition).active(0x6294230)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/memory_partition.go:156 +0x24
github.com/nakabonne/tstorage.(*storage).ensureActiveHead(0x615f980)
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/storage.go:346 +0x44
github.com/nakabonne/tstorage.(*storage).InsertRows.func1()
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/storage.go:295 +0x80
github.com/nakabonne/tstorage.(*storage).InsertRows(0x615f980, {0x63d28a0, 0x1, 0x1})
        /home/pi/go/pkg/mod/github.com/nakabonne/tstorage@v0.3.5/storage.go:324 +0x254
github.com/owncast/owncast/metrics.collectViewerCount()
        /home/pi/owncast/metrics/viewers.go:42 +0x14c
github.com/owncast/owncast/metrics.startViewerCollectionMetrics()
        /home/pi/owncast/metrics/viewers.go:25 +0x16c
created by github.com/owncast/owncast/metrics.Start
        /home/pi/owncast/metrics/metrics.go:69 +0x1a0
```

Looking at the stacktrace, I believe the error is caused by a known bug in the atomic package, where users must explicitly align 64-bit primitives on some 32-bit architectures: https://pkg.go.dev/sync/atomic#pkg-note-BUG

I've opened a PR on the tstorage module, which should resolve the issue: https://github.com/nakabonne/tstorage/pull/61

There doesn't appear to be any recent development on the module, so it's possible this bug will remain unaddressed. I'm mostly writing it up here as a reference, but don't expect any movement on Owncast's side.

**Comment by gabek** [comment]
Good find, and thanks for putting up that PR on their project! If there's no response I'll see if I can track down any forks that are active and could be used as drop-in replacements. Worst case we fork it and put this change in ourselves.

**Comment by mattdsteele** [comment]
Looks like the tstorage maintainers accepted my PR and released a new version, so once renovate picks up `0.3.6`, this should be fixed.

**Comment by gabek** [comment]
Wohoo! Thanks so much for pushing this through.

**Comment by gabek** [comment]
Closed in 4b97887751dfe40fb84f025718ae3c42b3459987!
